manager/volume.go

```diff
@@ -524,9 +526,26 @@ func (m *VolumeManager) Detach(name, nodeID string) (v *longhorn.Volume, err err
 func (m *VolumeManager) isVolumeAvailableOnNode(volume, node string) bool {
 	es, _ := m.ds.ListVolumeEngines(volume)
 	for _, e := range es {
-		if e.Spec.NodeID == node && e.Status.CurrentState == longhorn.InstanceStateRunning {
-			return true
+		if e.Spec.NodeID != node {
+			continue
+		}
+		if e.DeletionTimestamp != nil {
+			continue
+		}
+		if e.Spec.DesireState != longhorn.InstanceStateRunning || e.Status.CurrentState != longhorn.InstanceStateRunning {
+			continue
+		}
+		hasAvailableReplica := false
+		for _, mode := range e.Status.ReplicaModeMap {
+			if mode == longhorn.ReplicaModeRW {
+				hasAvailableReplica = true
+				break
+			}
```

Comment:
NIT: for small loops I like this approach, I think it makes it more readable, you also get rid of the conditional code.
```suggestion
			hasAvailableReplica = hasAvailableReplica || mode == longhorn.ReplicaModeRW
```